File: examples/c-sharp-examples/CollisionPrecise.dox.cs
using static MR;

public class CollisionPreciseExample
{
    public static void Run(string[] args)
    {
        var meshA = MR.makeUVSphere(); // make mesh A
        var meshB = MR.makeUVSphere(); // make mesh B
        meshB.transform(MR.AffineXf3f.translation(new MR.Vector3f(0.1f, 0.1f, 0.1f))); // shift mesh B for better demonstration

        var meshPartA = new MeshPart(meshA);
        var meshPartB = new MeshPart(meshB);

        var converters = MR.getVectorConverters(meshPartA, meshPartB).toInt; // create converters to integer field (needed for absolute precision predicates)
        var collidingFaceEdges = MR.findCollidingEdgeTrisPrecise(meshPartA, meshPartB, converters); // find each intersecting edge/triangle pair
        // print pairs of edges triangles
        for (ulong i = 0; i < collidingFaceEdges.size(); i++)
        {
            var vet = collidingFaceEdges[i];
            if (vet.isEdgeATriB())
                Console.WriteLine($"edgeA: {vet.edge.id}, triB: {vet.tri().id}");
            else
                Console.WriteLine($"triA: {vet.tri().id}, edgeB: {vet.edge.id}");
        }
    }
}

File: examples/cpp-examples/CollisionPrecise.dox.cpp
#include "MRMesh/MRMakeSphereMesh.h"
#include "MRMesh/MRMesh.h"
#include "MRMesh/MRMatrix3.h"
#include "MRMesh/MRAffineXf.h"
#include "MRMesh/MRMeshCollidePrecise.h"
#include <iostream>

int main()
{
    auto meshA = MR::makeUVSphere(); // make mesh A
    auto meshB = MR::makeUVSphere(); // make mesh B
    meshB.transform( MR::AffineXf3f::translation( MR::Vector3f( 0.1f, 0.1f, 0.1f ) ) ); // shift mesh B for better demonstration

    auto converters = MR::getVectorConverters( meshA, meshB ); // create converters to integer field (needed for absolute precision predicates)
    auto collidingFaceEdges = MR::findCollidingEdgeTrisPrecise( meshA, meshB, converters.toInt ); // find each intersecting edge/triangle pair 
    // print pairs of edges triangles
    for ( const auto& vet : collidingFaceEdges )
    {
        if ( vet.isEdgeATriB() )
            std::cout << "edgeA: " << int( vet.edge ) << ", triB: " << vet.tri().get() << "\n";
        else
            std::cout << "triA: " << vet.tri().get() << ", edgeB: " << int( vet.edge ) << "\n";
    }
    return 0;
}
